internal/tfschema/tfschema.go

```diff
@@ -110,41 +109,53 @@ func getTerragruntPluginPath(dir string) string {
 	return ret
 }
 
-func getClient(providerName string, dir string, iacType common.IACType) (tfschema.Client, error) {
+func getResourceSchema(resourceType string, dir string, iacType common.IACType) (*ResourceSchema, error) {
 	if iacType == common.Terragrunt {
 		// which mode of terragrunt it is (with or without cache folder).
 		if _, err := os.Stat(dir + "/.terragrunt-cache"); err == nil {
 			dir = getTerragruntPluginPath(dir)
 		}
 	}
 
-	providerToClientMapLock.Lock()
-	defer providerToClientMapLock.Unlock()
+	providerSchemasMapLock.Lock()
+	defer providerSchemasMapLock.Unlock()
 
-	client, exists := providerToClientMap[providerName]
-	if exists {
-		return client, nil
-	}
+	providerSchemas, ok := providerSchemasMap[dir]
+	if !ok {
+		providerSchemas = &ProviderSchemas{}
 
-	logger := hclog.New(&hclog.LoggerOptions{
-		Name:   "plugin",
-		Level:  hclog.Trace,
-		Output: hclog.DefaultOutput,
-		// this annoyance - both tfschema and go-plugin open output
-		// directly to os.Stderr, bypassing our log level filter.
-		// weird to need to bypass the issue by assigning the default
-		// output ¯\_(ツ)_/¯
-	})
+		cmd := exec.Command("terraform", "providers", "schema", "-json")
```

Comment:
Wondering if there's a better way to get the schema from the TF related packages?

Reply:
The required libraries have been moved to "internal". This is why we couldn't keep using tfschema with newer terraform versions.
E.g.: https://github.com/hashicorp/terraform/blob/main/internal/command/providers_schema.go

See https://github.com/hashicorp/terraform/pull/28723 (https://github.com/hashicorp/terraform/pull/28723#issuecomment-866075216)